drivers/ethernet/eth_mcux.c

```diff
@@ -391,52 +384,54 @@ void eth_mcux_phy_stop(struct eth_context *context)
 
 static void eth_mcux_phy_event(struct eth_context *context)
 {
-#if !(defined(CONFIG_ETH_MCUX_NO_PHY_SMI) && ETH_MCUX_FIXED_LINK)
-	uint32_t status;
-#endif
 	bool link_up;
-#if defined(CONFIG_SOC_SERIES_IMX_RT)
-	status_t res;
-	uint32_t ctrl2;
-#endif
 	phy_duplex_t phy_duplex = kPHY_FullDuplex;
 	phy_speed_t phy_speed = kPHY_Speed100M;
 
 #if defined(CONFIG_ETH_MCUX_PHY_EXTRA_DEBUG)
-	LOG_DBG("%s phy_state=%s", eth_name(context->base),
-		phy_state_name(context->phy_state));
+	LOG_DBG("%s phy_state=%s, link=%d", eth_name(context->base),
+		phy_state_name(context->phy_state), context->link_up);
 #endif
+
 	switch (context->phy_state) {
 	case eth_mcux_phy_state_initial:
 #if defined(CONFIG_SOC_SERIES_IMX_RT)
-		ENET_DisableInterrupts(context->base, ENET_EIR_MII_MASK);
-		res = PHY_Read(context->base, context->phy_addr,
-			       PHY_CONTROL2_REG, &ctrl2);
-		ENET_EnableInterrupts(context->base, ENET_EIR_MII_MASK);
-		if (res != kStatus_Success) {
-			LOG_WRN("Reading PHY reg failed (status 0x%x)", res);
-			k_work_submit(&context->phy_work);
+		if (context->phy_miface == PHY_MGMT_IF_SMI) {
+			uint32_t ctrl2;
+			status_t res;
+
+			context->phy_state = eth_mcux_phy_state_reset;
+			ENET_DisableInterrupts(context->base,
+					       ENET_EIR_MII_MASK);
+			res = PHY_Read(context->base, context->phy_addr,
+				       PHY_CONTROL2_REG, &ctrl2);
+			ENET_EnableInterrupts(context->base,
+					      ENET_EIR_MII_MASK);
+			if (res != kStatus_Success) {
+				LOG_WRN("Reading PHY reg failed (status 0x%x)",
+					res);
+				k_work_submit(&context->phy_work);
+			} else {
+				ctrl2 |= PHY_CTL2_REFCLK_SELECT_MASK;
+				ENET_StartSMIWrite(context->base,
+						   context->phy_addr,
+						   PHY_CONTROL2_REG,
+						   kENET_MiiWriteValidFrame,
+						   ctrl2);
+			}
 		} else {
-			ctrl2 |= PHY_CTL2_REFCLK_SELECT_MASK;
-			ENET_StartSMIWrite(context->base, context->phy_addr,
-					   PHY_CONTROL2_REG,
-					   kENET_MiiWriteValidFrame,
-					   ctrl2);
-		}
-		context->phy_state = eth_mcux_phy_state_reset;
-#endif
-#if defined(CONFIG_ETH_MCUX_NO_PHY_SMI)
-		/*
-		 * When the iface is available proceed with the eth link setup,
-		 * otherwise reschedule the eth_mcux_phy_event and check after
-		 * 1ms
-		 */
-		if (context->iface) {
-		       context->phy_state = eth_mcux_phy_state_reset;
+			/*
+			 * When the iface is available proceed with
+			 * the eth link setup, otherwise reschedule the
+			 * eth_mcux_phy_event and check after 1ms
+			 */
+			if (context->iface) {
+				context->phy_state = eth_mcux_phy_state_reset;
+			}
+			k_delayed_work_submit(&context->delayed_phy_work,
+					      K_MSEC(50));
```

Comment:
Why 50ms was it stressing the core at 1ms?

Reply:
I would not say that it is stressing the core but it can take some time (1 sec or so) to have the interface ready, therefore I think the 1 ms "polling" is unnecessary. With 1ms it also overflows the log if `CONFIG_ETH_MCUX_PHY_EXTRA_DEBUG` is set.